pycbc/transforms.py

```diff
@@ -1468,6 +1468,509 @@ def from_config(cls, cp, section, outputs):
+            latitude_ssb_param = self._default_latitude_ssb
+        if polarization_ssb_param is None:
+            polarization_ssb_param = self._default_polarization_ssb
+        self.tc_geo_param = tc_geo_param
+        self.longitude_geo_param = longitude_geo_param
+        self.latitude_geo_param = latitude_geo_param
+        self.polarization_geo_param = polarization_geo_param
+        self.tc_ssb_param = tc_ssb_param
+        self.longitude_ssb_param = longitude_ssb_param
+        self.latitude_ssb_param = latitude_ssb_param
+        self.polarization_ssb_param = polarization_ssb_param
+        self._inputs = [self.tc_geo_param, self.longitude_geo_param,
+                        self.latitude_geo_param, self.polarization_geo_param]
+        self._outputs = [self.tc_ssb_param, self.longitude_ssb_param,
+                         self.latitude_ssb_param, self.polarization_ssb_param]
+        super(GEOToSSB, self).__init__()
+
+    def transform(self, maps):
+        """This function transforms arrival time, sky localization,
+        and polarization angle in the geocentric frame to the corresponding
+        values in the SSB frame.
+
+        Parameters
+        ----------
+        maps : a mapping object
+
+        Returns
+        -------
+        out : dict
+            A dict with key as parameter name and value as numpy.array or float
+            of transformed values.
+        """
+        out = {}
+        out[self.tc_ssb_param], out[self.longitude_ssb_param], \
+            out[self.latitude_ssb_param], out[self.polarization_ssb_param] = \
+            coordinates.geo_to_ssb(
+                maps[self.tc_geo_param], maps[self.longitude_geo_param],
+                maps[self.latitude_geo_param], maps[self.polarization_geo_param]
+                )
+        return self.format_output(maps, out)
+
+    def inverse_transform(self, maps):
+        """This function transforms arrival time, sky localization,
+        and polarization angle in the SSB frame to the corresponding
+        values in the geocentric frame.
+
+        Parameters
+        ----------
+        maps : a mapping object
+
+        Returns
+        -------
+        out : dict
+            A dict with key as parameter name and value as numpy.array or float
+            of transformed values.
+        """
+        out = {}
+        out[self.tc_geo_param], out[self.longitude_geo_param], \
+            out[self.latitude_geo_param], out[self.polarization_geo_param] = \
+            coordinates.ssb_to_geo(
+                maps[self.tc_ssb_param], maps[self.longitude_ssb_param],
+                maps[self.latitude_ssb_param], maps[self.polarization_ssb_param]
+                )
+        return self.format_output(maps, out)
+
+    @classmethod
+    def from_config(cls, cp, section, outputs):
+        tag = outputs
+        skip_opts = []
+        additional_opts = {}
+
+        # get custom variable names
+        if cp.has_option("-".join([section, outputs]), 'tc-geo'):
```

Comment:
A lot of this code looks very repetitive and is doing the same type of thing at each line. Can't you compactify this by iterative over a list of the names rather than duplicating for each?